components/application-role-mgt/org.wso2.carbon.identity.application.role.mgt/src/main/java/org/wso2/carbon/identity/application/role/mgt/util/GroupIDResolver.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.role.mgt.util;
+
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.application.role.mgt.exceptions.ApplicationRoleManagementClientException;
+import org.wso2.carbon.identity.application.role.mgt.exceptions.ApplicationRoleManagementException;
+import org.wso2.carbon.identity.application.role.mgt.exceptions.ApplicationRoleManagementServerException;
+import org.wso2.carbon.identity.application.role.mgt.internal.ApplicationRoleMgtServiceComponentHolder;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * GroupId Resolver.
+ */
+public class GroupIDResolver implements IDResolver {
+
+    @Override
+    public String getNameByID(String id, String tenantDomain) throws ApplicationRoleManagementException {
+
+        String groupName = resolveGroupNameFromGroupID(id);
+        if (groupName == null) {
+            String errorMessage = "A group doesn't exist with id: " + id + " in the tenantDomain: " + tenantDomain;
+            throw new ApplicationRoleManagementClientException(errorMessage, errorMessage, "");
```

Comment:
Shall we properly throw the exception by adding the relevant details rather sending empty values?

Reply:
fixed